src/domains/Session/hooks/useVippsAuth/useVippsAuth.js

```diff
@@ -0,0 +1,80 @@
+import PATHS from 'pages/paths'
+import firebase from 'firebase/compat/app'
+import { notification } from 'antd'
+import { sendBackendRequest } from 'utils'
+import { transformUserDataFromVipps } from 'domains/Session/helpers'
+import { useCallback } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useTranslations } from 'contexts/Translation'
+
+// Vipps login part variables
+const APP_URL = process.env.REACT_APP_URL
+const { VIPPS_LOGIN_CALLBACK } = PATHS.UNAUTHENTICATED
+const AUTH_MARKER = 'auth'
+const VIPPS_AUTH_ENDPOINT = '/vipps/authorize'
+
+const useVippsAuth = () => {
+  const history = useHistory()
+  const { t } = useTranslations()
+
+  const vippsLogin = useCallback(
+    async (urlParamsObject) => {
+      const { code, state } = urlParamsObject
+      // For in app vipps auth add marker 'auth' to state
+      // to differentiate from widget auth and skip booking update
+      // if it just auth from login page
+      const [inAppAuthMarker] = state.split('_') //when it`s just auth state is in the form of auth_{generatedUserId}
+      const isAuth = inAppAuthMarker === AUTH_MARKER
+
+      const requestData = {
+        code: code,
+        redirectUri: APP_URL + VIPPS_LOGIN_CALLBACK
+      }
+      // Get access token and user data from vipps
+      const { data } = await sendBackendRequest({
+        endpoint: VIPPS_AUTH_ENDPOINT,
+        body: requestData,
+        errorDescription: t('Failed to get access token and data from vipps')
+      })
+
+      // If no token in response, it means that user didn`t authorize
+      if (!data?.token) {
+        notification.error({
```

Comment:
it's better to throw an error in this place, so u will have same error handling flow for all places/checks etc in catch section only (reduces amount of code u have to write).